eks-deployments/instance-domain-join.md

```diff
@@ -39,10 +39,13 @@ aws iam attach-role-policy --role-name $nodeInstanceRole --policy-arn arn:aws:ia
 aws ssm create-association --name $domainjoinSSMdoc --document-version 1 --targets "Key=tag:aws:autoscaling:groupName,Values=$autoScalingGroup"
 
 # Validate the association is created
+# The following command will show the status of command. If it is unsuccessful, you can
+# check the error message by checking the 'command history' under 'run command' in the
+# AWS Systems Manager Service
 aws ssm list-associations --association-filter-list "key=Name, value=$domainjoinSSMdoc"
 ```
 
-## 4. Create and join gMSA AD security group (Optional)
+## 4. Create and join gMSA AD security group
```

Comment:
could you explain why is this not optional anymore?

Reply:
This is not the security group we see in the instances, eg. SG-0... . It is the Active Directory security group which groups the gMSA accounts and has privileges associated with it. In my experience, it is necessary for the instances to join the security group else, it threw an error. So I made this step unoptional.

Reply:
The worker instance is added to the AD security group during AD domain join - https://github.com/aws-samples/amazon-eks-gmsa/blob/master/cloud-formation-templates/ssm-document-domain-join.yaml#L148

This step is only needed when the AD security group does not exist prior to domain join. This caution is already mentioned in the readme, which makes this step still optional.